### ISSUE TITLE
Coiniumserv Segwit Adoption - Litecoin Block Rejected

### ISSUE THREAD

**Issue by crusher72** [bonesoul/CoiniumServ#803]
Is CS going to adopt Segwit for Litecoin support? I mined a block of LTC 3 days ago and it was rejected. So sad...

Is the rejection due to the Segwit? Or was it a true reject?

Thanks,

Crusher72

**Comment by bonesoul** [comment]
We can accept a segwit PR.

**Comment by crusher72** [comment]
So this means it was a true reject?! Why would it reject? First time for me with a higher diff coin...

**Comment by bonesoul** [comment]
I can't comment on that without seeing the actual daemon data. Chances that you may have found an orphaned block where your pool though it was first in the network, where actually another was.

**Comment by BaalMcKloud** [comment]
ltc is using segwit and the code is not segwit ready i think.
can you post the daemon log for this block?

**Comment by crusher72** [comment]
Sorry can't post the log, the log was so large that I had to delete it to start fresh. It definitely is not segwit ready, when coiniumserv starts up it shows UPDATE REQUIRED, current version out of date.

**Comment by crusher72** [comment]
I will get a log if I can. But I just updated the LTC Wallet to the latest, and it says in the error log, that Coiniumserv needs to explicitly support Segwit. It does not currently, and gives errors.

**Comment by Virtuworks** [comment]
![image](https://user-images.githubusercontent.com/12708326/27257738-e4e14576-53af-11e7-98ed-23098b80c7a1.png)

Seems all that is needed to support Segwit on LTC.

**Comment by bonesoul** [comment]
@Virtuworks send a PR please so i can merge it in.

**Comment by Virtuworks** [comment]
Seems like your efforts are not with coinium anymore, I see Some new project you are putting a lot of time to that is clone of this one, why would we push something here if you aren't pushing your work here yourself?

**Comment by bonesoul** [comment]
to help the others and contribute the open source?

**Comment by Virtuworks** [comment]
Yes just wondering why your not doing the same. Is there a reason for a complete rewrite and start of a new project? We want to understand the future of this project or if we should be following something else.

**Comment by crusher72** [comment]
Thank you Virtuworks for your intelligent reply. Much appreciated. I second the question asked...what's going on bonesoul? We need some feedback on the direction of things so we all know what to do, contribute or just do enough to help ourselves on this titanic?

**Comment by Virtuworks** [comment]
I'm just wondering if we should be over here? 

https://github.com/bonesoul/hypeengine

Is this the right place to be?

**Comment by Virtuworks** [comment]
@crusher72 I've put two senior developers on the code so far, We've removed the built in Webserver and moved it to IIS, separated out the project into separate interface, dameon and other parts. We've fixed Bitcoin and Litecoin mining, along with the Hashrate display issues (they are related to old references in the code, we've also got fresh versions of all dependencies updated to latest versions. All this in 3 days, we are working on about 20 issues I've identified along with some deep improvements to the codebase. I'm not sure it's making much sense to share all that here, seems I'm already past anyone's contributions for the past 3 years if I did. I'm a strong open source advocate, I actually have some other projects that have been pretty successful but community is just about that, and I don't see much of that going on here. @crusher72 if you have a way to contact me somehow, happy to share binaries of what we've got done so far or at least hear from you what your experience is. I don't seem to find a way to make contact with someone.

**Comment by crusher72** [comment]
If you get this email, my email is dreamingofbrownie@hotmail.com

I would love the updated binaries so I don't have to manually compile it all again and make all these tweaks.

Let me know what you need done also and I can contribute when life allows me some time.

Best,

Crusher72


________________________________

**Comment by vagabondan** [comment]
@Virtuworks are you sure that adding only {"rules":["segwit"]} to getblocktemplate RPC parameter enables segwit support for CS?
It looks like coinbasetxn and blocktemplate should be modified.
Have you implemented all necessary changes already? Could you share them with us?

**Comment by Virtuworks** [comment]
![image](https://user-images.githubusercontent.com/12708326/28191357-cde8078a-67fd-11e7-9d9e-85d89d5f12c7.png)

Our fork is kind of a work in progress - You can see image - we have Litecoin and Bitcoin both working properly. We've made some major structural changes though and now are using StackExchange.Redis which was required to fix a LOCK bug that this code had. We've gotten past a lot of problems in this code. Being that we have now included a couple of commercial libraries to the code I need to see what the restrictions are for sharing. For instance this new theme is based on SmartAdmin. We are in the process of now adding Identity Framework and more of a Configured Worker / Worker Group as many other pools now use. I see there is a need for this type of app that works well, I will bring up to consider opening up our source to others. If so who would contribute? This source seems like it's been quite for a long time? Is there a benefit for posting changes here anymore? I can't seem to get an answer.

**Comment by Virtuworks** [comment]
![image](https://user-images.githubusercontent.com/12708326/28191488-9478763c-67fe-11e7-959d-e3242834e3c0.png)

**Comment by vagabondan** [comment]
Pictures show that blocks are calculated but block statistics for LTC is totally zero: no pending, no confirmed, no orphaned... Is this normal?

**Comment by Virtuworks** [comment]
@vagabondan this is a development site which is not in production and isn't actually earning coins right now, but its been tested and works.

**Comment by vagabondan** [comment]
@Virtuworks you've mentioned LOCK bug as the reason to migrate from csredis to StackExchange.Redis. I believe, I've met such a bug too: the problem is in synchronization of buffered stream (representing REDIS network connection). Did you try to modify csredis code before switching to StackExchange.Redis? Any hints?;)

Some details:
[Error] [CoiniumServ.Persistance.Layers.Hybrid.HybridStorage] [Litecoin] An exception occured while comitting share: Unexpected response type: Bulk instead of expected Int
stringified value:
3510662.859375
  at CSRedis.Internal.IO.RedisReader.ExpectType (CSRedis.RedisMessage expectedType) [0x00040] in /home/strannix/structure/projects/csredis/CSRedis/Inte
rnal/IO/RedisReader.cs:125 
  at CSRedis.Internal.IO.RedisReader.ReadInt (System.Boolean checkType) [0x00006] in /home/strannix/structure/projects/csredis/CSRedis/Internal/IO/Redi
sReader.cs:36 
  at CSRedis.Internal.Commands.RedisInt.Parse (CSRedis.Internal.IO.RedisReader reader) [0x00001] in /home/strannix/structure/projects/csredis/CSRedis/I
nternal/Commands/RedisInt.cs:13 
  at CSRedis.Internal.RedisConnector.Call[T] (CSRedis.RedisCommand`1[T] command) [0x00044] in /home/strannix/structure/projects/csredis/CSRedis/Interna
l/RedisConnector.cs:84 
  at CSRedis.RedisClient.Write[T] (CSRedis.RedisCommand`1[T] command) [0x00060] in /home/strannix/structure/projects/csredis/CSRedis/RedisClient.Sync.c
s:42 
  at CSRedis.RedisClient.HIncrBy (System.String key, System.String field, System.Int64 increment) [0x00001] in /home/strannix/structure/projects/csredi
s/CSRedis/RedisClient.Sync.cs:480 
  at CoiniumServ.Persistance.Layers.Hybrid.HybridStorage.AddShare (CoiniumServ.Shares.IShare share) [0x00071] in /home/strannix/structure/CoiniumServ/s
rc/CoiniumServ/Persistance/Layers/Hybrid/HybridStorage.Shares.cs:59

**Comment by Virtuworks** [comment]
@vagabondan yes we tried to fix CSRedis we couldn't find a way to get it to work without the LOCK added. With the LOCK added it would work, Stack REDIS works without the LOCKs added. When CSREdis is in use without the LOCKS, occasionally the site will stop working and we would start receiving REDIS errors. Once we changed to Stack REDIS all problems are non-existent.

**Comment by vagabondan** [comment]
@Virtuworks thank you! Is there any chance that you share this modification?:)
For now I gonna reinvent the wheel...

**Comment by Virtuworks** [comment]
@vagabondan we are considering it but haven't made a decision - if we did it wouldn't be here - based on how I've seen the existing source owner act I think it would be impossible for me to load all the changes we've made so far to project here. 

![image](https://user-images.githubusercontent.com/12708326/28316648-ac165c0a-6b91-11e7-840f-de239b2e5e55.png)

**Comment by Virtuworks** [comment]
I had misread, we are using this REDIS client which is working well and is free to distribute and use. 

https://stackexchange.github.io/StackExchange.Redis/

**Comment by bonesoul** [comment]
Last time i basically I asked you to send me a PR and told that I'd happily merge it. So please stop blaming me.

This is an open source project, everyone is free to share & contribute.

**Comment by Virtuworks** [comment]
Sorry? Fucking? I'm not following. 

I'm just making it clear it seems like this repository hasn't had much activity isn't that the case? 

Forcing you to do what? I don't see that anyone has requested anything from you except information about what your plans are nothing else and Ive stated my current experience with you. Now your cursing? I'm sorry but this repository doesn't seem friendly or willing to work with contributors, these are based on your actions man. 

Is asking you what your plans on this repository not allowed? 

I don't think I've blamed anyone specific for anything, I see 15 contributors on this repository not just you, like you said that means it belongs to everyone right? 

I'm sorry if I've angered you, and your right you have no requirement to do anything here so we just want to understand your intentions. Thx for making them clear!

**Comment by bonesoul** [comment]
I don't have any plans (and need to), this is an open source software as is with MIT license. Any one is free to use it or contribute it.

Basically I dont have any more time right now to continue development on this repository, as I'm basically the one that developed 99% of this software, I'm just maintaining the PR's.

So basically stop asking me my intentions or plans. Even if I develop an alternative implementation with node.js within my free time, I've free will to do so. 

This repository is more then happy to accept PR's and work with contributors as you can see from the git history.  But as in your case, you asked me to make a change to source code where I did requested a PR instead -- which is the way open source software world works.

So if you are not happy with it as is, you are free to fork your own repository and rule it as you like.

**Comment by Virtuworks** [comment]
@bonesoul Thanks for explanation, I never asked any help or for your to change anything at all - at least I cant remember doing that. I am well aware of how Open source works and am familiar as well with the feeling you have to tell me this - I manage another project and have managed some larger open source projects for many many years - i have a project now that has probably 40,000 users using it - I can completely relate to the constant requests from others to fix things and not understanding how open source works. Thanks I get your position!

**Comment by bonesoul** [comment]
Okay, I'm sorry then if we had a misunderstanding,  I'm happy if we are all settled now.

**Comment by bonesoul** [comment]
As a better explanation, I don't run any pools any more, that is the number 1 reason I'm not actively contributing the development right now. The other project you mentioned (https://github.com/bonesoul/hypepool) is just a toy project to practice with node.js.

**Comment by Virtuworks** [comment]
Understood, as you say and I fully agree, you have no requirement to do anything at all and are free to come here and do whatever you want whenever you want and don't need to explain that to anyone - we are 100% on the same page with this. But, as an open source leader for a long time, community needs to understand the intentions of the projects owner, your position is  well understood to me based on my own experiences. We appreciate all the work you've done so far, I get for the most part it's thankless work. Thank you!

**Comment by vagabondan** [comment]
For those who need: CoiniumServ version that was migrated from CSRedis to StackExchange.Redis is here:
https://github.com/vagabondan/CoiniumServ
Seems to work. But could be (and should be later) improved obviously.

**Comment by TnTBass** [comment]
@vagabondan I can't install your version of CoiniumServ, I submitted an issue on your project:
https://github.com/vagabondan/CoiniumServ/issues/1

**Comment by vagabondan** [comment]
@TnTBass I've answered + see new commit with packages

**Comment by cryptocoincollector** [comment]
What is the proper procedure for posting an issue and perhaps finding answers to this. I paid for the compiled version but now I have my doubts.
